### PR TITLE
Add ORCBlockOutputFormat

### PR DESCRIPTION
**Pull request by Avogar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add ORCBlockOutputFormat

Detailed description / Documentation draft:
### ORC output format
This output format writes data in orc format.

### CODE REVIEW

**Comment by Avogar** [comment]
I forgot to add include path in cmake, I'l add it later

File: src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+    number_orc_column->resize(number_column.size());
+
+    for (size_t i = 0; i != number_column.size(); ++i)
+    {
+        if (null_bytemap && (*null_bytemap)[i])
+        {
+            number_orc_column->notNull[i] = 0;
+            continue;
+        }
+        if constexpr (std::is_same<NumberType, UInt8>::value)
```
Comment (nikitamikhaylov) [inline src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp:149]: I think it is dirty hack :) Please write comment about this special case.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -364,6 +364,12 @@ target_include_directories (clickhouse_common_io SYSTEM BEFORE PUBLIC ${DOUBLE_C
 
 target_include_directories (clickhouse_common_io SYSTEM BEFORE PUBLIC ${MSGPACK_INCLUDE_DIR})
 
+target_include_directories (clickhouse_common_io SYSTEM BEFORE PUBLIC ${ORC_INCLUDE_DIR})
```
Comment (nikitamikhaylov) [inline src/CMakeLists.txt:367]: Maybe better to write it under ` if (USE_ORC)` statement?

**Review by nikitamikhaylov — APPROVED** [review]
Please write more comments and add performance test (just include ORC format into existing tests of formats).

File: src/Processors/Formats/Impl/ORCBlockOutputFormat.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#pragma once
+
```
Comment (nikitamikhaylov) [inline src/Processors/Formats/Impl/ORCBlockOutputFormat.h:2]:
```suggestion
#include "config_formats.h"
#if USE_ORC
```

File: src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+        }
+        const StringRef & string = string_column.getDataAt(i);
+        string_orc_column->data[i] = const_cast<char *>(string.data);
+        string_orc_column->length[i] = string.size;
+    }
+    string_orc_column->numElements = string_column.size();
+}
+
+template <typename ColumnType, typename GetSecondsFunc, typename GetNanosecondsFunc>
+void ORCBlockOutputFormat::ORCBlockOutputFormat::writeDateTimes(
```
Comment (nikitamikhaylov) [inline src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp:208]: Double `ORCBlockOutputFormat::ORCBlockOutputFormat` ?

File: src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp (hunk 1 above)
Comment (nikitamikhaylov) [reply src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp:149]: And it doesn't work https://clickhouse-builds.s3.yandex.net/11662/717f63923a19b9a4b0932e6295ed60d3753e7009/build_log_707272824_1592914485.txt
Maybe add `Int8` to if statement?

**Comment by Avogar** [comment]
@nikitamikhaylov can you help? I can't understand what is wrong with Functional stateless tests (unbundled) and Performance test. They don't see ORC output format.

**Comment by nikitamikhaylov** [comment]
> @nikitamikhaylov can you help? I can't understand what is wrong with Functional stateless tests (unbundled) and Performance test. They don't see ORC output format.

I will help, of course. But there is another strange error with clang-tidy build... Which I don’t know how to fix.

**Comment by nikitamikhaylov** [comment]
Tests `01308_orc_output_format_arrays` and `01307_orc_output_format` are now disabled for unbundled build. Because tests for orc input format are already disabled.

**Comment by nikitamikhaylov** [comment]
Build failure is caused by internal CI problem.

**Pull request merged**